File: java-code/src/main/java/refactoring/conditional_logic/replace_conditional_with_polymorphism/example1/after/DefaultBird.java
package refactoring.conditional_logic.replace_conditional_with_polymorphism.example1.after;


public class DefaultBird extends Bird {

  public DefaultBird(Bird bird) {
    super(bird.getName(), bird.getType(), bird.getNumberOfCoconuts(), bird.getVoltage(), bird.isNailed());
  }

  @Override
  public String plumage() {
    return "unknown";
  }

  @Override
  public int airSpeedVelocity() {
    return 0;
  }
}

File: go-code/conditional_logic/replace_conditional_with_polymorphism/example1/after/DefaultBird.go
package after

type DefaultBird struct {
	BirdAttributes
}

func (n DefaultBird) GetBirdAttributes() BirdAttributes {
	return n.BirdAttributes
}

func (n DefaultBird) Plumage() string {
	return "unknown"
}

func (n DefaultBird) AirSpeedVelocity() int {

	return 0
}
